src/plugins/vis_type_table/public/components/table_vis_component_group.test.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import { TableVisComponentGroup } from './table_vis_component_group';
+import { TableVisConfig, ColumnSort } from '../types';
+import { Table, TableGroup } from '../table_vis_response_handler';
+
+jest.mock('./table_vis_component', () => ({
+  TableVisComponent: () => <div data-test-subj="TableVisComponent">TableVisComponent</div>,
+}));
+
+const table1 = {
+  table: {
+    columns: [],
+    rows: [],
+    formattedColumns: [],
+  } as Table,
+  title: '',
+} as TableGroup;
+
+const table2 = {
+  table: {
+    columns: [],
+    rows: [],
+    formattedColumns: [],
+  } as Table,
+  title: '',
+} as TableGroup;
+
+const tableUiStateMock = {
+  sort: { colIndex: undefined, direction: undefined } as ColumnSort,
+  setSort: jest.fn(),
+  width: [],
+  setWidth: jest.fn(),
+};
+
+describe('TableVisApp', () => {
+  it('should not render table component if no table', () => {
```

Comment:
```suggestion
  it('should not render table or table group components if no table', () => {
```
looks like this is what we're asserting? That neither is rendered?

Reply:
will update